### PR TITLE
uicache: Add support for user apps, fix app plugins on >=14.5, proper…

### PR DESCRIPTION
**Pull request by opa334**
…ly support group containers and containerization flags, improve code

This upstreams the changes from TrollStore to support User apps. I also did a few refactors such as directly linking MobileContainerManager.framework, improving the output of `-i`, added a new option `-s` for specifing that an app on the user partition should be registered as `System` and improved the code (a lot).

### CODE REVIEW

File: uicache.m [hunk 1]
```diff
@@ -118,267 +150,401 @@ void help() {
...
+
+		if (verbose) {
+			printf("Registering dictionary: %s\n", dictToRegister.description.UTF8String);
 		}
-		[plist setObject:bundlePlugins forKey:@"_LSBundlePlugins"];
-		if (![workspace registerApplicationDictionary:plist]) {
-			fprintf(stderr, _("Error: Unable to register %s\n"), path);
+
+		if (![workspace registerApplicationDictionary:dictToRegister]) {
+			printf("Error: Unable to register %s\n", path.fileSystemRepresentation);
```
Comment (CRKatri) [inline uicache.m:436]: Use `fprintf()` and keep it on `stderr`.

File: uicache.m [hunk 2]
```diff
@@ -118,267 +150,401 @@ void help() {
...
 		}
-	} else {
+	}
+	else {
+		NSURL *url = [NSURL fileURLWithPath:path];
 		if (![workspace unregisterApplication:url]) {
-			fprintf(stderr, _("Error: Unable to unregister %s\n"), path);
+			printf("Error: Unable to unregister %s\n", path.fileSystemRepresentation);
```
Comment (CRKatri) [inline uicache.m:442]: Use `fprintf()` and keep it on `stderr`.

**Review by CRKatri — CHANGES_REQUESTED** [review]
Some initial changes, I'll give a more thorough review later.
Could you document `-s` in the manpage too?

**Comment by opa334** [comment]
I unfortunately have no clue how to edit the manpage file. You will need to do it, also remove the caveat at the bottom about uicache not being able to operate on App Store apps.

Here is a description for the -s flag:

When registering an app inside `/var/containers`, it will normally be registered as a user app.
When this flag is set, the app will be registered as a system app, regardless of whether it resides inside `/var/containers` or not.

**Comment by CRKatri** [comment]
Will this work correctly on iOS 12+?

**Comment by opa334** [comment]
I don't see why it wouldn't, can test it

**Review by Shaflsharif — APPROVED** [review]

**Review by CRKatri — APPROVED** [review]

**Pull request merged**